arch/arm/src/arm/Toolchain.defs

```diff
@@ -92,7 +92,7 @@ endif
 
 CC = $(CROSSDEV)gcc
 CXX = $(CROSSDEV)g++
-CPP = $(CROSSDEV)gcc -E
+CPP = $(CROSSDEV)gcc -E -P -x c
```

Comment:
should this be `-x c++`

Reply:
@xiaoxiang781216 Did you have any thoughts on this?

Reply:
Since C preprocessor is same as C++ preprocessor, so I think both will be fine. But -x option is important if we want preprocess some non C/C++ file(e.g. link script or bash script).

Reply:
sounds good since this is what it was before as well.  There were some differences with RISC-V ABI in GCC, but it looks like those were identified as a bug and resolved.  I'll merge.